### PR TITLE
[NFC] Mitigate pointless copies

### LINKED ISSUE #95036: llvm/include/llvm/Frontend/OpenMP/OMPIRBuilder.h:201: Pointless copy of object ?
Static analyser cppcheck says:

llvm/include/llvm/Frontend/OpenMP/OMPIRBuilder.h:201:46: performance: Function parameter 'RHS' should be passed by const reference. [passedByValue]

Source code is

  bool operator<(const TargetRegionEntryInfo RHS) const {

Suggest new code:

  bool operator<(const TargetRegionEntryInfo & RHS) const {

### PR DESCRIPTION
**Pull request by bradenhelmer**
Fixes #95036 #95033 #94933 #94930

### CODE REVIEW

**Review by jh7370 — APPROVED** [review]
LGTM, pending pre-commit CI completing. I'd add `[NFC]` to the start of the PR title.

Do you need me to click the merge button?

**Comment by bradenhelmer** [comment]
> LGTM, pending pre-commit CI completing. I'd add `[NFC]` to the start of the PR title.
> 
> 
> 
> Do you need me to click the merge button?

Yeah, thanks

**Pull request merged**
